Jq transformer (#2411)
Kong/docs.konghq.com: app/_hub/tibbycat/jq-transformer/index.md

---
name: jq-transformer
publisher: tibbycat
categories:
  - transformations
type: plugin
desc: Extends kong with a wrapper for the amazing ./jq JSON processor
description: |
  Trivially transforms your request or response body with the https://stedolan.github.io/jq/ processor if the JSON payloads are slightly mis-aligned, without modifying code in your end service.
  * Transform the request & response to simplify the interface to your end service (eg Confluent's kafka REST proxy)
  * Remove sensitive(/private) information from your response payload.
  * Reduce your response payload size.

>   NB: Content-Type header will probably affect your results. (application/.*json is probably the safest)

Reviewer: ```suggestion

  <div class="alert alert-warning">
    <strong>Note: </strong> The Content-Type header will probably affect your results (`application/.*json` is probably the safest).
  </div>
```

# Versions are categorized by Kong edition and their known compatibility.
# Unlisted Kong versions will be considered incompatible.
# Uncomment at least one of 'community_edition' or 'enterprise_edition'.
# Add array-formatted lists of versions under their appropriate subsection.

kong_version_compatibility:
  community_edition:
    compatible:
      - 0.12.x
      - 0.14.x
#    incompatible:
  enterprise_edition:
    compatible:
      - 1.3.x
      - 1.5.x
      - 2.1.x
#    incompatible:

# EXAMPLE kong_version_compatibility block: this example shows how to indicate
# various compatibilities. Also see other extension files in _app/_hub/ for more examples.
#kong_version_compatibility:
#  community_edition:
#    compatible:
#      - 0.12.x
#      - 0.13.x
#      - 0.14.x
#  enterprise_edition:
#    compatible:
#      - 0.32-x
#      - 0.33-x
#      - 0.34-x

#########################
# PLUGIN-ONLY SETTINGS below this line
# If your plugin is an extension of the core platform, ALL of the following
# lines must be completed.
# If NOT defined as a 'plugin' in line 32, delete all lines up to '# BEGIN MARKDOWN CONTENT'

params: # Metadata about your plugin
  name: jq-transformer

>   service_id:

Reviewer: ```suggestion
  service_id: true 
```

    # Boolean - whether this plugin can be applied to a Service.
    # Affects generation of examples and config table.
  consumer_id:
    # Boolean - whether this plugin can be applied to a Consumer.
    # Affects generation of examples and config table.

>   route_id:

Reviewer: ```suggestion
  route_id: true
```

    # whether this plugin can be applied to a Route.
    # Affects generation of examples and config table.
  protocols:
    # List of protocols this plugin is compatible with, in array format.
    # Valid values: "http", "https", "tcp", "tls"
    ["http", "https"]

>   dbless_compatible:

Reviewer: IS this plugin compatible with db-less mode? k8? Please see tabs for OSS and Enterprise here https://docs.konghq.com/hub/plugins/compatibility/

Author: I am unsure of this as I don’t use dB-less mode. However, I don’t explicitly interact with the dB. And commonly use decK to apply the plugin.

Author: Would it be acceptable to say ‘partially’ as this ‘could be’ but is ‘untested’?

Reviewer: Probably not but I'll defer to @eskibars

    # Degree of compatibility with DB-less mode. Three values allowed:
    # 'yes', 'no' or 'partially'.
  dbless_explanation:
    # Optional free-text explanation, usually containing details about the degree of
    # compatibility with DB-less.
  yaml_examples:
    # Boolean - enables or disables autogenerated examples in declarative YAML
    # format. Default is `true`; set to `false` to disable only declarative
    # YAML examples.
  k8s_examples:
    # Boolean - enables or disables autogenerated examples in Kubernetes YAML
    # format. Default is `true`; set to `false` to disable only K8s examples.
  examples:
    # Boolean - enables or disables all autogenerated examples (cURL, YAML, and
    # Kubernetes). Default is `true`; set to `false` to disable all
    # autogenerated examples.

  config: # Configuration settings for your plugin
    - name: Request
      required: no
      default: empty string
      value_in_examples:

>         # If the field is to appear in examples, this is the value to use.

Reviewer: Can you provide examples for request & response strings?

    # compatibility with DB-less.
  yaml_examples:
    # Boolean - enables or disables autogenerated examples in declarative YAML
    # format. Default is `true`; set to `false` to disable only declarative
    # YAML examples.
  k8s_examples:
    # Boolean - enables or disables autogenerated examples in Kubernetes YAML
    # format. Default is `true`; set to `false` to disable only K8s examples.
  examples:
    # Boolean - enables or disables all autogenerated examples (cURL, YAML, and
    # Kubernetes). Default is `true`; set to `false` to disable all
    # autogenerated examples.

  config: # Configuration settings for your plugin
    - name: Request
      required: no
      default: ""
      value_in_examples:
        {"custom":"Additional custom value for upstream service"}+.
      description:
        ./jq expression to transform request body by
    - name: Response
      required: no
      default: ""
      value_in_examples:
        {"custom":"Additional custom value for client"}+.
      description:
        ./jq expression to transform response body by
  #  - name: # add additional setting blocks as needed, each demarcated by -
  extra:

>     If you wish to remove sensitive data. For example {"name":"tibbycat","sex":"male"}, you could transform this by the expression {"name":.name}

Reviewer: ```suggestion
    If you want to remove sensitive data such as {"name":"tibbycat","sex":"male"}, you could transform this using the expression {"name":.name}
```
